File: laravel/remove_watermark.py
#!/usr/bin/env python
import os
import sys
try:
    from PIL import Image
except ImportError:
    try:
        import Image
    except:
        print("PIL not found, trying to install...")
        os.system('pip install pillow')
        from PIL import Image

def remove_watermark(input_path):
    try:
        img = Image.open(input_path)
        width, height = img.size
        
        # Watermark region (bottom-right 15%)
        wm_x_start = int(width * 0.85)
        wm_y_start = int(height * 0.85)
        wm_width = width - wm_x_start
        wm_height = height - wm_y_start
        
        # Sample colors from left of watermark
        sample_width = min(80, wm_x_start)
        total_r, total_g, total_b, count = 0, 0, 0, 0
        
        for x in range(max(0, wm_x_start - sample_width), wm_x_start):
            for y in range(wm_y_start, min(height, wm_y_start + wm_height)):
                r, g, b = img.getpixel((x, y))
                total_r += r
                total_g += g
                total_b += b
                count += 1
        
        if count > 0:
            avg_r = total_r // count
            avg_g = total_g // count
            avg_b = total_b // count
            
            # Fill watermark area with average color
            for x in range(wm_x_start, width):
                for y in range(wm_y_start, height):
                    img.putpixel((x, y), (avg_r, avg_g, avg_b))
        
        # Save with same format
        img.save(input_path)
        return True
    except Exception as e:
        print(f"Error processing {input_path}: {str(e)}")
        return False

def main():
    input_dir = r"C:\xampp\htdocs\goprint\laravel\storage\app\public\products"
    
    print("=" * 60)
    print("GoPrint Image Watermark Remover")
    print("=" * 60)
    print(f"\nProcessing directory: {input_dir}\n")
    
    files = []
    for f in os.listdir(input_dir):
        if f.lower().endswith(('.png', '.jpg', '.jpeg')):
            files.append(os.path.join(input_dir, f))
    
    print(f"Found {len(files)} images to process")
    print("-" * 60)
    
    success = 0
    failed = 0
    
    for i, filepath in enumerate(files, 1):
        filename = os.path.basename(filepath)
        print(f"[{i}/{len(files)}] Processing: {filename}")
        
        if remove_watermark(filepath):
            print("   SUCCESS")
            success += 1
        else:
            print("   FAILED")
            failed += 1
    
    print("-" * 60)
    print(f"\nCompleted!")
    print(f"Success: {success}")
    print(f"Failed: {failed}")

if __name__ == "__main__":
    main()

File: laravel/remove_watermark.php
<?php
/**
 * GoPrint Image Watermark Remover
 * PHP version - works with Laravel environment
 */

set_time_limit(300);

$inputDir = 'C:\\xampp\\htdocs\\goprint\\laravel\\storage\\app\\public\\products';

echo "========================================\n";
echo "GoPrint Image Watermark Remover\n";
echo "========================================\n\n";

echo "Processing directory: $inputDir\n\n";

$files = glob($inputDir . '/*.{png,jpg,jpeg}', GLOB_BRACE);

echo "Found " . count($files) . " images to process\n";
echo "----------------------------------------\n\n";

$success = 0;
$failed = 0;

foreach ($files as $file) {
    $filename = basename($file);
    echo "Processing: $filename\n";
    
    try {
        $info = getimagesize($file);
        $width = $info[0];
        $height = $info[1];
        $mime = $info['mime'];
        
        switch ($mime) {
            case 'image/jpeg':
                $img = imagecreatefromjpeg($file);
                break;
            case 'image/png':
                $img = imagecreatefrompng($file);
                break;
            default:
                echo "   SKIPPED (unsupported format)\n";
                continue 2;
        }
        
        // Watermark region (bottom-right 15%)
        $wm_x_start = (int)($width * 0.85);
        $wm_y_start = (int)($height * 0.85);
        $wm_width = $width - $wm_x_start;
        $wm_height = $height - $wm_y_start;
        
        // Sample colors from left of watermark
        $sample_width = min(80, $wm_x_start);
        $total_r = $total_g = $total_b = $count = 0;
        
        for ($x = max(0, $wm_x_start - $sample_width); $x < $wm_x_start; $x++) {
            for ($y = $wm_y_start; $y < min($height, $wm_y_start + $wm_height); $y++) {
                $rgb = imagecolorat($img, $x, $y);
                $r = ($rgb >> 16) & 0xFF;
                $g = ($rgb >> 8) & 0xFF;
                $b = $rgb & 0xFF;
                $total_r += $r;
                $total_g += $g;
                $total_b += $b;
                $count++;
            }
        }
        
        if ($count > 0) {
            $avg_r = (int)($total_r / $count);
            $avg_g = (int)($total_g / $count);
            $avg_b = (int)($total_b / $count);
            
            $fill_color = imagecolorallocate($img, $avg_r, $avg_g, $avg_b);
            
            // Fill watermark area
            imagefilledrectangle($img, $wm_x_start, $wm_y_start, $width, $height, $fill_color);
        }
        
        // Save back to original file
        switch ($mime) {
            case 'image/jpeg':
                imagejpeg($img, $file, 90);
                break;
            case 'image/png':
                imagepng($img, $file);
                break;
        }
        
        imagedestroy($img);
        
        echo "   SUCCESS\n";
        $success++;
        
    } catch (Exception $e) {
        echo "   FAILED: " . $e->getMessage() . "\n";
        $failed++;
    }
}

echo "\n----------------------------------------\n";
echo "Completed!\n";
echo "Success: $success\n";
echo "Failed: $failed\n";
echo "========================================\n";
?>
